internal/handlers/tigris/json_schema.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tigris
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/common"
+	"github.com/FerretDB/FerretDB/internal/tjson"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+)
+
+// getJSONSchema returns a marshaled JSON schema received from validator -> $jsonSchema.
+func getJSONSchema(doc *types.Document) (*tjson.Schema, error) {
+	v, err := common.GetRequiredParam[*types.Document](doc, "validator")
+	if err != nil {
+		return nil, err
+	}
+
+	schema, err := common.GetRequiredParam[*types.Document](v, "$jsonSchema")
+	if err != nil {
+		return nil, err
+	}
+
+	return schemaFromDocument(schema)
+}
+
+// schema creates a new TJSON Schema based on the types.Document format.
+// The given doc should contain the keys typical for schema (e.g. title, type etc).
+func schemaFromDocument(doc *types.Document) (*tjson.Schema, error) {
```

Comment:
Do we need both that function and `tjson.DocumentSchema`? Can we merge them?

If not, comments for both functions should explain the difference

Reply:
Actually, I guess that's a version of `tjson.Schema.Unmarshal`. But do we need both? `Unmarshal` unmarshals into struct with `DisallowUnknownFields` that does most of this code already

Reply:
Added a comment. If this is not enough, please help me reformulate it.
I'd love to have a raw JSON or whatever else instead of `types.Document`, but I didn't find how to do it.

Reply:
> I'd love to have a raw JSON or whatever else instead of types.Document

I think we should threat Tigris' collection schema as types.Document – it is a JSON object, and types.Document represents BSON object with all its data types, fields order, etc

Reply:
This is the exact thing I stuck with last week, so if you help me understand how to do it, that would be great. Scheduled a pair programming session for it.